example/openapi/api-docs.yaml

```diff
@@ -1035,331 +1361,458 @@ components:
-      type: object
-      properties:
-        username:
-          type: string
-        password:
-          type: string
-    JwtAuthenticationResponse:
+    RefJson:
+      required:
+      - name
+      - type
       type: object
       properties:
-        token:
+        status:
           type: string
-    CommitJson:
-      type: object
-      properties:
-        source:
+          readOnly: true
+        description:
           type: string
-        comment:
+        parentRefId:
           type: string
+          default: master
         deleted:
-          type: array
-          items:
-            type: object
-            additionalProperties:
-              type: object
-        added:
-          type: array
-          items:
-            type: object
-            additionalProperties:
-              type: object
-        updated:
-          type: array
-          items:
-            type: object
-            additionalProperties:
-              type: object
-        type:
+          type: boolean
+          readOnly: true
+        parentCommitId:
           type: string
         name:
           type: string
         id:
           type: string
+        type:
+          type: string
+          enum:
+          - Branch
+          - Tag
+        _projectId:
+          type: string
+          readOnly: true
         _refId:
           type: string
+          readOnly: true
         _creator:
           type: string
+          readOnly: true
         _docId:
           type: string
-        _projectId:
-          type: string
+          readOnly: true
         _modifier:
           type: string
-        _modified:
-          type: string
+          readOnly: true
         _created:
           type: string
-        _commitId:
+          readOnly: true
+        _modified:
           type: string
-      additionalProperties:
-        type: object
-    CommitsResponse:
+          readOnly: true
+      additionalProperties: true
+    RefsResponse:
+      required:
+      - refs
       type: object
       properties:
         messages:
           type: array
+          nullable: true
           items:
             type: string
+            nullable: true
         rejected:
           type: array
+          nullable: true
           items:
             $ref: '#/components/schemas/Rejection'
-        commits:
+        refs:
           type: array
           items:
-            $ref: '#/components/schemas/CommitJson'
-    CommitsRequest:
+            $ref: '#/components/schemas/RefJson'
+    RefsRequest:
+      required:
+      - refs
       type: object
       properties:
         source:
           type: string
+          nullable: true
         comment:
           type: string
-        commits:
+          nullable: true
+        refs:
           type: array
           items:
-            $ref: '#/components/schemas/CommitJson'
-    ProjectJson:
+            $ref: '#/components/schemas/RefJson'
+    CommitAddedJson:
```

Comment:
Can we merge the `Commit + (Added|Updated|Deleted) + Json` models? None of the properties are specified as required, so CommitUpdatedJson (renamed) can be used for all of them.

Reply:
changed to ElementUpdated